pkg/plugins/scorer/no_hit_lru.go

```diff
@@ -0,0 +1,263 @@
+	return &coldRequestState{isCold: c.isCold}
+}
+
+// NoHitLRUFactory defines the factory function for the NoHitLRU
+func NoHitLRUFactory(name string, rawParameters json.RawMessage, handle plugins.Handle) (plugins.Plugin, error) {
+	parameters := NoHitLRUParameters{}
+	if rawParameters != nil {
+		if err := json.Unmarshal(rawParameters, &parameters); err != nil {
+			return nil, fmt.Errorf("failed to parse the parameters of the '%s' scorer - %w", NoHitLRUType, err)
+		}
+	}
+
+	if parameters.PrefixPluginName == "" {
+		parameters.PrefixPluginName = prefix.PrefixCachePluginType
+	}
+
+	// Note: We don't enforce that the prefix plugin exists here
+	// The scorer will gracefully handle missing prefix cache state as an optimization
+
+	return NewNoHitLRU(handle.Context(), &parameters).WithName(name), nil
+}
+
+// NewNoHitLRU creates a new NoHitLRU scorer
+func NewNoHitLRU(ctx context.Context, params *NoHitLRUParameters) *NoHitLRU {
+	prefixPluginName := prefix.PrefixCachePluginType
+	lruSize := defaultLRUSize
+
+	if params != nil {
+		if params.PrefixPluginName != "" {
+			prefixPluginName = params.PrefixPluginName
+		}
+		if params.LRUSize > 0 {
+			lruSize = params.LRUSize
+		}
+	}
+
+	lruCache, err := lru.New[string, struct{}](lruSize)
+	if err != nil {
+		log.FromContext(ctx).Error(err, fmt.Sprintf("failed to initialize NoHitLRU scorer: could not create LRU cache with size %d: %v", lruSize, err))
+		return nil
+	}
+
+	return &NoHitLRU{
+		typedName:        plugins.TypedName{Type: NoHitLRUType},
+		lruCache:         lruCache,
+		mutex:            &sync.RWMutex{},
+		prefixPluginName: prefixPluginName,
+		pluginState:      plugins.NewPluginState(ctx),
+	}
+}
+
+// NoHitLRU scorer that favors pods that were least recently used for cold requests.
+// This can help evenly distribute cache growth, since cold requests result in more
+// new KV blocks.
+type NoHitLRU struct {
+	typedName        plugins.TypedName
+	lruCache         *lru.Cache[string, struct{}] // pod name -> dummy value (we only care about order)
+	mutex            *sync.RWMutex
+	prefixPluginName string
+	pluginState      *plugins.PluginState
+}
+
+// TypedName returns the typed name of the plugin.
+func (s *NoHitLRU) TypedName() plugins.TypedName {
+	return s.typedName
+}
+
+// WithName sets the name of the plugin.
+func (s *NoHitLRU) WithName(name string) *NoHitLRU {
+	s.typedName.Name = name
+	return s
+}
+
+// Score scores the given pods based on LRU for cold requests.
+// For cache hits, returns neutral scores (0.5) for all pods.
+// For cache misses, ranks pods by their LRU order.
+// - LRU ordering is with respect to when a pod last received a cold request.
+// - Least recently used (or never used) pods get highest score (1.0)
+// - Most recently used pods get lowest score (approaching 0.0)
+func (s *NoHitLRU) Score(ctx context.Context, cycleState *types.CycleState, request *types.LLMRequest, pods []types.Pod) map[types.Pod]float64 {
```

Comment:
I think it might help readability if the main logical blocks in this call are separated to different functions.